# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.cpp
# Commit: 415c7e1ea941
# Full Hash: 415c7e1ea941ca3bc0b33c5d91fe9782044ee9a0
# Author: Timothy Nikkel <tnikkel@gmail.com>
# Date: 2022-02-12 21:25:59
# Regressor Bug: 1754436
# File Overlap Count: 2
# Description:
#   Bug 1754662. Refactor nsLayoutUtils::GetPopupFrameForEventCoordinates. r=emilio
#   
#   This lets us re-use code for bug 1754436.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138392
# ==============================================================================

diff -r 152c2969ca06 -r 415c7e1ea941 accessible/generic/LocalAccessible.cpp
--- a/accessible/generic/LocalAccessible.cpp	Sat Feb 12 04:34:08 2022 -0500
+++ b/accessible/generic/LocalAccessible.cpp	Sat Feb 12 09:41:50 2022 +0000
@@ -511,13 +511,10 @@
 
   LayoutDeviceIntRect rootRect = rootWidget->GetScreenBounds();
 
-  WidgetMouseEvent dummyEvent(true, eMouseMove, rootWidget,
-                              WidgetMouseEvent::eSynthesized);
-  dummyEvent.mRefPoint =
-      LayoutDeviceIntPoint(aX - rootRect.X(), aY - rootRect.Y());
-
-  nsIFrame* popupFrame = nsLayoutUtils::GetPopupFrameForEventCoordinates(
-      accDocument->PresContext()->GetRootPresContext(), &dummyEvent);
+  auto point = LayoutDeviceIntPoint(aX - rootRect.X(), aY - rootRect.Y());
+
+  nsIFrame* popupFrame = nsLayoutUtils::GetPopupFrameForPoint(
+      accDocument->PresContext()->GetRootPresContext(), rootWidget, point);
   if (popupFrame) {
     // If 'this' accessible is not inside the popup then ignore the popup when
     // searching an accessible at point.